osf/models/metaschema.py

```diff
@@ -50,14 +50,19 @@ def get_latest_version(self, name):
         """
         return self.filter(name=name).order_by('schema_version').last()
 
-    def get_latest_versions(self, request=None, invisible=False):
+    def get_latest_versions(self, request=None, invisible=False, default_schema_id=None):
         """
         Return the latest version of the given schema
 
         :param request: the request object needed for waffling
         :return: queryset
         """
-        queryset = self.order_by('name', '-schema_version').distinct('name')
+        annotated_qs = self.annotate(default_schema_ordering=models.Case(
+            models.When(id=default_schema_id, then=models.Value(1)),
+            default=models.Value(0),
+            output_field=models.PositiveIntegerField(),
+        ))
+        queryset = annotated_qs.order_by('name', '-schema_version', 'default_schema_ordering').distinct('name')
```

Comment:
This `order_by` isn't quite right. We need to order by `default_schema_ordering` first and in descending order (or else violate truth-y behavior by setting to 0 if the ids equal and 1 otherwise).